tests/postgres_tests/test_schemas.py

```diff
@@ -0,0 +1,23 @@
+from django.test import modify_settings
+
+from . import PostgreSQLTestCase
+from .models import SchemaModel
+
+
+@modify_settings(INSTALLED_APPS={'append': 'django.contrib.postgres'})
+class SchemaTest(PostgreSQLTestCase):
+
+    Model = SchemaModel
+
+    def setUp(self):
+        self.Model.objects.bulk_create([
+            self.Model(int1=1),
+            self.Model(int1=2),
+            self.Model(int1=3),
+        ])
+
+    def test_values(self):
+        index = 1
+        for object in self.Model.objects.all():
+            self.assertEqual(object.int1, index)
+            index += 1
```

Comment:
I can modify the tests to use a schema name, but with this approach that requires duplicating the test cases.  The simple requirement for this fix is that you should be able to address tables in a non-default schema.  It seems to me that it's a pre-cursor to  #6162. Which edge cases are you referring to?  I'm no SQL or PostgreSQL guru, but to my knowledge you should be able to use fully qualified table names anywhere in SQL where you can use tables.